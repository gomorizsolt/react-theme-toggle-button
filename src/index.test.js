import React from "react";
import "@testing-library/jest-dom";
import renderer from "react-test-renderer";
import { fireEvent, render, screen } from "@testing-library/react";
import ReactThemeToggleButton from "./index";

describe("ReactThemeToggleButton", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(<ReactThemeToggleButton isDark onChange={() => {}} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  describe("when the `isDark` prop is true", () => {
    it("sets the label's title and aria-label attributes to `Activate light mode`", () => {
      const { getByTitle } = render(
        <ReactThemeToggleButton isDark onChange={() => {}} />
      );

      expect(getByTitle(/activate/i).title).toEqual("Activate light mode");
      expect(getByTitle(/activate/i).getAttribute("aria-label")).toEqual(
        "Activate light mode"
      );
    });
  });

  describe("when the `isDark` prop if false", () => {
    it("sets the label's title and aria-label attributes to `Active dark mode`", () => {
      const { getByTitle } = render(
        <ReactThemeToggleButton isDark={false} onChange={() => {}} />
      );

      expect(getByTitle(/activate/i).title).toEqual("Activate dark mode");
      expect(getByTitle(/activate/i).getAttribute("aria-label")).toEqual(
        "Activate dark mode"
      );
    });
  });

  describe("when the toggle button is pressed", () => {
    let onChangeSpy;

    beforeEach(() => {
      onChangeSpy = jest.fn();
    });

    afterEach(() => {
      onChangeSpy.mockReset();
    });

    it("fires the passed function", () => {
      const { getByTitle } = render(
        <ReactThemeToggleButton isDark onChange={onChangeSpy} />
      );

      fireEvent.click(getByTitle(/activate/i));

      expect(onChangeSpy).toHaveBeenCalled();
    });
  });

  describe("when `invertIconLogic` prop is true", () => {
    it("inverts icon logic", () => {
      render(
        <ReactThemeToggleButton isDark onChange={() => {}} invertedIconLogic />
      );

      expect(screen.getByRole("checkbox")).not.toBeChecked();
    });
  });
});
