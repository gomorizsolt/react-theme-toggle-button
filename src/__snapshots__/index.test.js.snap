// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReactThemeToggleButton renders correctly 1`] = `
<label
  aria-label="Activate light mode"
  className="container"
  title="Activate light mode"
>
  <input
    checked={true}
    onChange={[Function]}
    type="checkbox"
  />
  <div />
</label>
`;
